File: Task41/app/src/main/java/edu/deakin/s600152989/sit305/task41/TaskAdapter.java
package edu.deakin.s600152989.sit305.task41;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.text.BreakIterator;
import java.util.ArrayList;
import java.util.List;

public class TaskAdapter extends RecyclerView.Adapter<TaskAdapter.TaskViewHolder> {

    private List<Task> tasks;
    private OnTaskClickListener listener;

    public TaskAdapter(List<Task> tasks, OnTaskClickListener listener) {
        this.tasks = tasks;
        this.listener = listener;
    }

    @NonNull
    @Override
    public TaskViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_task, parent, false);
        return new TaskViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull TaskViewHolder holder, int position) {
        Task currentTask = tasks.get(position);
        holder.textViewTitle.setText(currentTask.getTitle());
        holder.textViewDescription.setText(currentTask.getDescription());
        holder.textViewDueDate.setText("Due: " + currentTask.getDueDate());

        // Handle Edit Button click
        holder.editButton.setOnClickListener(v -> {
            if (listener != null) {
                listener.onEditClick(currentTask);
            }
        });

        // Handle Delete Button click
        holder.deleteButton.setOnClickListener(v -> {
            if (listener != null) {
                listener.onDeleteClick(currentTask);
            }
        });
    }

    @Override
    public int getItemCount() {
        return tasks.size();
    }

    public void setTasks(List<Task> taskList) {
        this.tasks = taskList;
        notifyDataSetChanged();
    }

    public Task getTaskAt(int position) {
        return tasks.get(position);
    }

    static class TaskViewHolder extends RecyclerView.ViewHolder {
        private final TextView textViewTitle;
        private final TextView textViewDescription;
        private final TextView textViewDueDate;
        private final Button editButton;
        private final Button deleteButton;  // Add delete button

        public TaskViewHolder(View itemView) {
            super(itemView);
            this.textViewTitle = itemView.findViewById(R.id.textViewTitle);
            this.textViewDescription = itemView.findViewById(R.id.textViewDescription);
            this.textViewDueDate = itemView.findViewById(R.id.textViewDueDate);
            this.editButton = itemView.findViewById(R.id.edit_button);
            this.deleteButton = itemView.findViewById(R.id.delete_button); // Initialize delete button
        }
    }

    // Interface for handling task click events
    public interface OnTaskClickListener {
        void onEditClick(Task task);
        void onDeleteClick(Task task);
    }
}

File: Task41/app/src/main/java/edu/deakin/s600152989/sit305/task41/TaskAdapter.kt
package edu.deakin.s600152989.sit305.task41

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import edu.deakin.s600152989.sit305.task41.TaskAdapter.TaskViewHolder

class TaskAdapter(private var tasks: List<Task>, private val listener: OnTaskClickListener?) :
    RecyclerView.Adapter<TaskViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TaskViewHolder {
        val itemView = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_task, parent, false)
        return TaskViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: TaskViewHolder, position: Int) {
        val currentTask = tasks[position]
        holder.textViewTitle.text = currentTask.title
        holder.textViewDescription.text = currentTask.description
        holder.textViewDueDate.text = "Due: " + currentTask.dueDate

        // Handle Edit Button click
        holder.editButton.setOnClickListener { v: View? ->
            listener?.onEditClick(currentTask)
        }

        // Handle Delete Button click
        holder.deleteButton.setOnClickListener { v: View? ->
            listener?.onDeleteClick(currentTask)
        }
    }

    override fun getItemCount(): Int {
        return tasks.size
    }

    fun setTasks(taskList: List<Task>) {
        this.tasks = taskList
        notifyDataSetChanged()
    }

    fun getTaskAt(position: Int): Task {
        return tasks[position]
    }

    class TaskViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textViewTitle: TextView =
            itemView.findViewById(R.id.textViewTitle)
        val textViewDescription: TextView =
            itemView.findViewById(R.id.textViewDescription)
        val textViewDueDate: TextView =
            itemView.findViewById(R.id.textViewDueDate)
        val editButton: Button =
            itemView.findViewById(R.id.edit_button)
        val deleteButton: Button =
            itemView.findViewById(R.id.delete_button) // Add delete button
        // Initialize delete button
    }

    // Interface for handling task click events
    interface OnTaskClickListener {
        fun onEditClick(task: Task)
        fun onDeleteClick(task: Task?)
    }
}